lib/state_machine/state_machine.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "consts.hpp"
+#include "state.cpp"
+#include "transition.cpp"
+
+#include <optional>
+
+namespace hyped::state_machine {
+
+class StateMachine {
+ public:
+  StateMachine() { current_state = State::kInitialState; };
+
+  void transition(const Message message);
+
+  std::optional<Message> checkTransition();
+
+  void reset();
```

Comment:
This is not defined in your set of transitions so how do you define this behavior?